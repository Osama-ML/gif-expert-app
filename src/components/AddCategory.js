import React, { useState } from 'react'

export const AddCategory = ({setCategories}) => {

    const [InputValue, setInputValue] = useState('Hola mundo');

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        setCategories(categ => [...categ, InputValue]);
    }

    return (
        <>
            <form onSubmit={handleSubmit}>
                <input 
                type="text"
                value={InputValue}
                onChange={handleInputChange}
                />
            </form>
        </>
    )
}
