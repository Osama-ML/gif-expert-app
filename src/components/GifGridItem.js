import React from 'react'

export const GifGridItem = ({title, url}) => {

    console.log(title)

    return (
        <>
            <div>
                <hr/>
                <p>{title}</p>
                <img src={url} alt={title}></img>
            </div>
        </>
    )   
}
