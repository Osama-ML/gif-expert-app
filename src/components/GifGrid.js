import React, { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';


export const GifGrid = ({category}) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs(category)
            .then(gifs => setImages(gifs))
    }, [category]) // se ejecueta cuando el componente es renderifazo por primera vez
    // si la dependencia category cambia, se vuelve a ejecutar el useEffect



    return (
        <>
            <h3>
                {category}
            </h3>
            <div className="card-grid">
                {
                    images.map((img) => {
                        return(<GifGridItem key= {img.id} 
                            { ...img } /> // se puede enviar datos de esta maanera tmb
                        )
                        })
                }
            </div>
        </>
    )
}
