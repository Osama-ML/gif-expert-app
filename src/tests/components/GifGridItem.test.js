import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en <GifGridItem/>', () => {

    const titulo = 'TÃ­tulo';
    const url = 'https://localhost/algo.jpeg';

    test('Debe de mostrar el contenido correctamente', () => {
        
        const wrapper = shallow(<GifGridItem title={titulo} url={url}/>)
        expect(wrapper).toMatchSnapshot();
    })
    
})
