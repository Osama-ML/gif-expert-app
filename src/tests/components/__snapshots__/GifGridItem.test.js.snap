// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe de mostrar el contenido correctamente 1`] = `
<Fragment>
  <div
    className="card animate__animated animate__fadeIn"
  >
    <img />
    <p />
  </div>
</Fragment>
`;
